[EngSys] Get Vcpkg automatically (#3614)
Azure/azure-sdk-for-cpp: CONTRIBUTING.md

1. From the Azure SDK GitHub repo, click on the "Code -> Open with Codespaces" button.
1. Open a Terminal. The development environment will be ready for you. Continue to [Building and Testing](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#building-and-testing).

### VS Code Codespaces

1. Install the [VS Code Remote Extension Pack](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack)
1. When you open the Azure SDK for C++ repo in VS Code, it will prompt you to open the project in the Dev Container. If it does not prompt you, then hit CTRL+P, and select "Remote-Containers: Open Folder in Container..."
1. Open a Terminal. The development environment will be ready for you. Continue to [Building and Testing](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#building-and-testing).

## Full Local Setup

> ### ** Pre-requisites **

Reviewer: nit: consider removing these custom formatting

> #### -> CMake

Reviewer: I don't think we need to highlight headings like this. Consistent formatting makes the doc easier to read and navigate.

```suggestion
#### CMake
```

CMake version 3.13 or higher is required to build these libraries. Download and install CMake from the project's
[website](https://cmake.org/download/).

### Third Party Dependencies

> Azure SDK use Vcpkg manifest mode to declare the [list of required dependencies](https://github.com/Azure/azure-sdk-for-cpp/blob/main/vcpkg.json) for building the SDK service libraries. It will also get and set up Vcpkg automatically. You can move on to [Building the project](#building-the-project) and skip next part if you are not interested in learning the alternatives for setting up dependencies.

Reviewer: The relevant part to the "quick" reader, of just ignore this step, should be a bit more clear.

```suggestion
Azure SDK uses Vcpkg manifest mode to declare the [list of required 3rd party dependencies](https://github.com/Azure/azure-sdk-for-cpp/blob/main/vcpkg.json) for building the SDK service libraries. It will also get and set up Vcpkg automatically. **You can move on to [Building the project](#building-the-project)** and skip the next part if you are not interested in learning about alternatives for setting up dependencies.
```

> If you didn't jump to `Build the project`, you will now learn about how you can control Vcpkg and dependencies integration.

Reviewer: nit: Trying to read through this, to understand what action I need to take and when. It seems a bit challenging to follow the flow of the doc, so snippets like these seem unnecessary.

This text could be the title of the next section.

Maybe remove this sentence.
```suggestion
```

> #### Setting up Vcpkg automatic integration

Reviewer: ```suggestion
#### Customize the Vcpkg dependency integration
```

> If the CMake option _-DCMAKE_TOOLCHAIN_FILE=..._ is not defined to generate the project, the Azure SDK project will automatically get Vcpkg and link it to get its dependencies. You can use the next environment variables to change this behavior:

Reviewer: nit: Am I supposed to define the `DCMAKE_TOOLCHAIN_FILE` option if I want to customize the vcpkg integration? How am I as a contributor supposed to see if this is defined or being used or not.

The table below doesn't mention that, so you may want to keep the info about how VCPKG_ROOT interacts with `CMAKE_TOOLCHAIN_FILE` on behalf of the user.

> <center>

Reviewer: nit: Can we use typical markdown table format instead of HTML-tags? They don't seem necessary here.

Like this:
Environment Variable | Description
-- | --
AZURE_SDK_DISABLE_AUTO_VCPKG | When defined, Vcpkg won't be automatically cloned and linked. Use this setting, for example, if your dependencies are installed in the system and you don't need to get them.
AZURE_SDK_VCPKG_COMMIT | This variable can set the git commit id to be used when automatically cloning Vcpkg.
VCPKG_ROOT | Use this variable to set an existing Vcpkg folder from your system to be linked for building. Use this, for example, when working with Vcpkg classic mode, to switch between different Vcpkg folders.

Author: It's usually easier (to me) reading html tables than markdown tables.  
But I'll make the change. I don't mind

<table>
<tr>
<td>Environment Variable</td>
<td>Description</td>
</tr>
<tr>
<td>AZURE_SDK_DISABLE_AUTO_VCPKG</td>
<td>When defined, Vcpkg won't be automatically cloned and linked. Use this setting, for example, if your dependencies are installed in the system and you don't need to get them.</td>
</tr>
<tr>
<td>AZURE_SDK_VCPKG_COMMIT</td>
<td>This variable can set the git commit id to be used when automatically cloning Vcpkg.</td>
</tr>
<tr>
<td>VCPKG_ROOT</td>

> <td>Use this variable to set an existing Vcpkg folder from your system to be linked for building. Use this, for example, when working with Vcpkg classic mode, to switch between different Vcpkg folders.</td>

Reviewer: It's interesting to use "system" to refer to the developer machine or computer. I suspect most folks are using a desktop machine (whether local or remote) to contribute to the repo, so "machine" or "computer" felt more relevant to the reader.

That said, I suppose, if we include VMs, codespaces, and other such environments, a general term like system could work better.

</tr>
</table>

</center>

> ## Building the project

Reviewer: How long does building the entire repo take? How long does the "first" build take? It would be helpful to give that info to first time contributors as a heads-up so they know what to expect.

We should add some detail about building using targets as well given our repo is growing and contributors may not want to build the entire repo.
